lib/gfx/gx_shader.cpp

```diff
@@ -863,8 +863,18 @@ wgpu::ShaderModule build_shader(const ShaderConfig& config, const ShaderInfo& in
           var dist_attn = dot(light.dist_att, vec3f(1.0, dist, dist2));
           attn = max(0.0, cos_attn / dist_attn);)""");
       } else if (cc.attnFn == GX_AF_SPEC) {
-        diffFn = GX_DF_NONE;
-        FATAL("AF_SPEC unimplemented");
+        std::string normal = UsePerPixelLighting ? "in.mv_nrm" : "mv_nrm";
+        lightAttnFn = fmt::format(R"""(
+          var cosine = select(0.0, max(0.0, dot({0}, light.dir)), dot({0}, ldir) >= 0.0);
+          var cos_attn = dot(light.cos_att, vec3f(1.0, attn, attn * attn));
+          var dist_attn = 0.0;
+          if ({1}) {{
```

Comment:
Would be cool to write different paths instead of putting true/false in the shader